doc/kci_testsuite.md

```diff
@@ -0,0 +1,300 @@
+        └── v4l2-parser.sh
+
+```
+   `/usr/bin/v4l2-parser.sh` will be available in the `buster-v4l2` rootfs image. These overlay scripts are normally run from LAVA job to execute some tests and parse the results.
+    This sample [LAVA job definition](https://lava.collabora.co.uk/scheduler/job/2446126/definition#defline120) shows how `v4l2-parser.sh` can be used.
+
+
+6. Hopefully now you should have some basic idea about how to add a test suite to a rootfs and how it can be used in LAVA environment.
+
+### How to add new testsuite to a buster rootfs image
+
+For this example, we will be using `btrfs-progs` test-sute.
+
+1. Let's first modify `rootfs-configs.yaml` for `buster-btrfs` with appropriate entires. 
+
+
+```
+  buster-btrfs:
+    rootfs_type: debos
+    debian_release: buster
+    arch_list:
+      - amd64
+    extra_packages:
+      - git 
+      - autoconf 
+      - automake 
+      - gcc 
+      - make 
+      - pkg-config 
+      - e2fslibs-dev 
+      - libblkid-dev 
+      - zlib1g-dev 
+      - liblzo2-dev 
+      - asciidoc 
+      - xmlto 
+      - libzstd-dev
+      - python3-setuptools
+      - python3.7-dev
+    script: "scripts/buster-btrfs.sh"
+    test_overlay: "overlays/btrfs"
+```
+
+
+2. We will look into last three config entries namely, `extra_packages`, `script` and `test_overlay`.
+   `extra_packages` as name suggests it will add the listed packages from final rootfs image.
+
+3. Remember `script` path is relative to `kci_rootfs build --data-path` value. By default its `jenkins/debian/debos/script`
+   Here we need add new script named `buster-btrfs.sh`. Its a simple script to download the repo and build and install 
+   the btrfs-progs from source. 
+
+
+```
+$ cat jenkins/debian/debos/script/buster-btrfs.sh
+
+git clone https://github.com/kdave/btrfs-progs.git /btrfs-progs
+cd /btrfs-progs
+./autogen.sh && ./configure
+make
+make install
+```
+
+
+4. `test_overlay` option is used to create specific directory layout inside the rootfs image. In our case, 
+
+```
+$ tree jenkins/debian/debos/overlays/btrfs/
+jenkins/debian/debos/overlays/btrfs/
+└── usr
+    └── bin
+        └── btrfs-verify-cli.sh
+```
+
+   `/usr/bin/btrfs-verify-cli.sh` will be available on `buster-btrfs` rootfs image. These overlay scripts normally ran from LAVA job.
+
+```
+$ cat jenkins/debian/debos/overlays/btrfs/btrfs-verify-cli.sh
+
+cd  /btrfs-progs && TEST=001\* ./tests/cli-tests.sh
+```
+
+5. Make sure execute permission is applied on the scripts. 
+
+```
+chmod +x jenkins/debian/debos/overlays/btrfs/usr/bin/btrfs-verify-cli.sh
+chmod +x jenkins/debian/debos/script/buster-btrfs.sh
+```
+
+6. Now build `buster-btrfs` rootfs image using kci_rootfs. 
+
+7. Once rootfs image is created, you can use it with appropriate kernel image. Make sure that kernel config has 
+   appropriate btrfs configs and other file system specific entries enabled. 
+
+8. Add appropriate test entry with LAVA job definition for `btrfs-verify-cli.sh`. For example,
```

Comment:
```suggestion
8. Add appropriate test entry to the LAVA job definition in order to run `btrfs-verify-cli.sh`. For example,
```

Reply:
About numbered list, I wonder if it's possible here to use the automatic Markdown mode by using `1.` for each item.  It might get reset to 1 with the extra formatted things in-between, but if this can work it will be easier to make changes in the future.

Reply:
right, if that works, its easier to make changes in later, this will be fine for someone who reads using browser. But for users who cloned this repo, it may be little inconvenient if they use text editors or commands like `less`. I think we can skip adding automatic markdown `1.` for now?

Reply:
OK that's not a big deal, we can leave it as-is for now.  I think automatic numbering really makes sense for small lists, but this is a bit more complicated.